From f2e32117838c0b095c9789ac74597ee3b0f3c13d Mon Sep 17 00:00:00 2001
From: Andrea della Porta <andrea.porta@suse.com>
Date: Fri, 6 Feb 2026 16:54:32 +0100
Subject: [PATCH] dts: bcm2712: Extend PCIe range to encompass shared SRAM

The actual DT nodes from PCIe root complex to the RP1 endpoint
device do not include the sram shared memory range. While this
shared memory is not used yet, there's an error from mmio-sram
driver which cannot map the entire range.

Extend the range accordingly.

Signed-off-by: Andrea della Porta <andrea.porta@suse.com>
---
 arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts  | 2 +-
 arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi | 2 +-
 arch/arm64/boot/dts/broadcom/bcm2712.dtsi         | 2 +-
 arch/arm64/boot/dts/broadcom/rp1.dtsi             | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts b/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts
index bd6f747..1ed6b4a 100644
--- a/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts
@@ -140,7 +140,7 @@ rp1_target: &pcie2 {
 	// This is the RP1 peripheral space
 	ranges = <0x00 0x40000000
 		  0x01 0x00 0x00000000
-		  0x00 0x00400000>;
+		  0x00 0x00410000>;
 
 	dma-ranges =
 	// inbound RP1 1x_xxxxxxxx -> PCIe 1x_xxxxxxxx
diff --git a/arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi b/arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi
index beff401..c3dc847 100644
--- a/arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi
@@ -127,7 +127,7 @@ rp1_target: &pcie2 {
 	// This is the RP1 peripheral space
 	ranges = <0x00 0x40000000
 		  0x01 0x00 0x00000000
-		  0x00 0x00400000>;
+		  0x00 0x00410000>;
 
 	dma-ranges =
 	// inbound RP1 1x_xxxxxxxx -> PCIe 1x_xxxxxxxx
diff --git a/arch/arm64/boot/dts/broadcom/bcm2712.dtsi b/arch/arm64/boot/dts/broadcom/bcm2712.dtsi
index 58d8790..e5f7cb3 100644
--- a/arch/arm64/boot/dts/broadcom/bcm2712.dtsi
+++ b/arch/arm64/boot/dts/broadcom/bcm2712.dtsi
@@ -552,7 +552,7 @@
 
 			dma-ranges =
 				/* 4MiB, 32-bit, non-prefetchable at PCIe 00_0000_0000 */
-				<0x02000000 0x00 0x00000000 0x1f 0x00000000 0x00 0x00400000>,
+				<0x02000000 0x00 0x00000000 0x1f 0x00000000 0x00 0x00410000>,
 				/* 64GiB, 64-bit, prefetchable at PCIe 10_0000_0000 */
 				<0x43000000 0x10 0x00000000 0x00 0x00000000 0x10 0x00000000>,
 				/* 4KiB, 64-bit, non-prefetchable at PCIe ff_ffff_f000 MIP0 */
diff --git a/arch/arm64/boot/dts/broadcom/rp1.dtsi b/arch/arm64/boot/dts/broadcom/rp1.dtsi
index a401aae..4411ae5 100644
--- a/arch/arm64/boot/dts/broadcom/rp1.dtsi
+++ b/arch/arm64/boot/dts/broadcom/rp1.dtsi
@@ -9,7 +9,7 @@
 	#size-cells = <2>;
 	ranges = <0x01 0x00 0x00000000
 		  0x02000000 0x00 0x00000000
-		  0x0 0x400000>;
+		  0x0 0x410000>;
 	interrupt-controller;
 	#interrupt-cells = <2>;
 
-- 
2.51.0

